modules/apps/headless/headless-builder/headless-builder-impl/src/main/java/com/liferay/headless/builder/internal/publisher/APIApplicationPublisherImpl.java

```diff
@@ -113,7 +163,12 @@ private void _unregisterServiceRegistration(
 
 	private static BundleContext _bundleContext;
 
-	private final Map<String, ServiceRegistration<Application>>
-		_headlessBuilderApplicationServiceRegistrationMap = new HashMap<>();
+	private final Map
+		<String,
+		 ObjectValuePair
```

Comment:
To be honest, I don't know if I like the pair in this situation :cry: but it is ok for me, it is just an internal thought. What about a List of `ServiceRegistration`?

Reply:
Done